paasta_tools/deployd/watchers.py

```diff
@@ -45,6 +55,130 @@ def startup_checker(self, obj):
         self.is_ready = True
 
 
+class PublicConfigFileWatcher(PaastaWatcher):
+
+    def __init__(self, inbox_q, cluster, **kwargs):
+        super(PublicConfigFileWatcher, self).__init__(inbox_q, cluster)
+        self.wm = pyinotify.WatchManager()
+        self.wm.add_watch(PATH_TO_SYSTEM_PAASTA_CONFIG_DIR, self.mask, rec=True)
+        self.notifier = pyinotify.Notifier(watch_manager=self.wm,
+                                           default_proc_fun=PublicConfigEventHandler(filewatcher=self))
+
+    @property
+    def mask(self):
+        boring_flags = ['IN_CLOSE_NOWRITE', 'IN_OPEN', 'IN_ACCESS', 'IN_ATTRIB']
+        return reduce(lambda x, y: x | y,
+                      [v for k, v in pyinotify.EventsCodes.OP_FLAGS.items() if k not in boring_flags])
+
+    def run(self):
+        self.notifier.loop(callback=self.startup_checker)
+
+    def startup_checker(self, obj):
+        self.is_ready = True
+
+
+def get_marathon_apps():
+    marathon_config = load_marathon_config()
+    marathon_client = get_marathon_client(marathon_config.get_url(), marathon_config.get_username(),
```

Comment:
Hmm, I guess quite a bit: at most once per 20s but iff a host has been marked for draining since the last run. I'll think about whether we can live with slightly stale data here and if we can I'll try and import the cached client...